### PR TITLE
Retire Position assignment operator

### PR DESCRIPTION
**Pull request by mcostalba**
Assorted formatting and comment tweaks in position.h

Also retired advanced_pawn_push(), it just hides the logic in
the only two places where it is used: better to make the rule
explicit to allow easier tweaks and tests.

No functional change.

### CODE REVIEW

**Comment by Karlon** [comment]
Looks good.

**Comment by zamar** [comment]
Don't you need to disable the assigment operator?

**Comment by mcostalba** [comment]
Yes you are right...and doing so we have a compile error because assignment
is used here in start_thinking():

RootPos = pos;

And because it comes very naturally to use assignment in that context, I
think I will reintroduce assignment operator and update the pull request :-(

**Comment by mcostalba** [comment]
I haev updated teh pull request changing also the description.

**Comment by zamar** [comment]
After brining back the assignment operator, it's just about code reshuffling, and it looks fine fine for me.

approved.

File: src/search.cpp [hunk 1]
```diff
@@ -1214,7 +1215,8 @@ namespace {
           && !InCheck
           && !givesCheck
           &&  futilityBase > -VALUE_KNOWN_WIN
-          && !pos.advanced_pawn_push(move))
```
Comment (glinscott) [inline src/search.cpp]: Why remove these?  Inlining these conditions hurts readability here IMO.  advanced_pawn_push is a good name for what the function does.

**Comment by glinscott** [comment]
I made one comment about the advanced_pawn_push removal, but otherwise, looks good.

**Comment by mcostalba** [comment]
Ok, I have re-added  advanced_pawn_push and committed the patch. Thanks.

**Pull request closed without merging**